docs/end-user/filtering-call-trees.md

```diff
@@ -0,0 +1,81 @@
+
+The following are the different types of filtering operations that are supported.
+
+| Filter type | Description |
+| ----------- | ----------- |
+| Search filter | Drop samples that do not match a text string. |
+| Implementation filter | Restrict stacks to an implementation—native (C++) stacks, or to JavaScript stacks. |
+| Invert call stack | Flip the sample's stacks upside down, and build a new call tree. |
+| Transforms | Modify the shape of the call tree according to some operation. Typically this only modifies the stacks. |
+
+## Search filter
+
+![A screenshot of the search box](../images/search-2018-05.jpg)
+
+Searching will exclude samples that do not match a search string. The search filter looks at every function in the sample's stack. If the function's name, resource, or URL match, then the sample will be retained, otherwise it will be filtered out. Multiple search terms can be added, as long as they are separated by commas.
+
+![The image demonstrates the search filter. On the left is a call tree that has all stacks, and on the right is the filtered call tree that does not have any stacks that do not have the function C.](../images/filter-search.svg)
+
+The above diagram is reproduced in the profile below:
+
+* Before searching: [https://perfht.ml/2I3SMsR](https://perfht.ml/2I3SMsR)
+* After searching: [https://perfht.ml/2rbcj0N](https://perfht.ml/2rbcj0N)
+
+The following are some ideas on how to use search terms:
+
+ * `js::` - Filter for a C++ namespace.
+ * `www.example.com` - Filter for a domain name
+ * `www.example.com/assets/scripts.js` - Filter for a single script.
+
+## Implementation filter
+
+![A screenshot of the implementation filter dropdown.](../images/implementation-2018-05.jpg)
+
+The implementation filter is useful for narrowing down the call tree to a subset of stack types. For instance it can be useful to only show native stacks (C++ and Rust) or to only show JavaScript stacks.
+
+The following graphic demonstrates how it can be useful to filter to only JavaScript stacks.
+
+![](../images/implementation-filter.svg)
+
+Before the the implementation filter, the call tree contains both C++ JavaScript internals, as well as the JavaScript stacks. This interleaving could be useful when diagnosing how the JS engine is optimizing code, but for normal JS performance profiling this could be confusing to find hot JS functions. When the native stacks are removed, the call tree that is generated is much more sensible for representing the execution of JavaScript code.
+
+The implementation filter will modify the shape of stacks, and generate a completely new call tree. If any of the filtered stacks are empty, then those samples will be dropped from the analysis.
```

Comment:
Optional: I don't understand properly the second sentence here: what's empty exactly ?

Reply:
Ok, I clarified this a bit.